shapes/profile/app.shex

```diff
@@ -0,0 +1,36 @@
+PREFIX oidc: <http://www.w3.org/ns/solid/oidc#>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX schem: <http://schema.org/>
+PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX acl: <http://www.w3.org/ns/auth/acl#>
+PREFIX cert:  <http://www.w3.org/ns/auth/cert#>
+PREFIX ldp: <http://www.w3.org/ns/ldp#>
+PREFIX sp: <http://www.w3.org/ns/pim/space#>
+PREFIX solid: <http://www.w3.org/ns/solid/terms#>
+
+<#id> {
+  a schem:SoftwareApplication
+    // rdfs:comment  "A software application. (from Schema.org)" ;
+  oidc:redirect_uris IRI + 
+    // rdfs:comment  "URIs where the Solid Identity Provider may redirect the user to complete the login process" ;
+  oidc:scope xsd:string ?
+    // rdfs:comment  "A string containing a space-delimited list of OAuth2.0 scopes your application is allowed to request." ;
+  oidc:grant_types [ "authorization_code"
+                      "refresh_token" ] +
```

Comment:
there _could_ be other grant types that a server supports. `authorization_code` is the minimum required scope for Solid, but other scopes could be present. I would use the `EXTRA` keyword here, and I wouldn't mention `refresh_token`, since that depends on `authorization_code` being present:

```
<#id> EXTRA oidc:grant_types {
    ...
    oidc:grant_types [ "authorization_code" ]
}
```